src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -209,6 +218,12 @@ internal abstract class AbstractPullDiagnosticHandler<TDiagnosticsParams, TRepor
                     return new ProjectOrDocumentId(project.Id);
                 }
 
+                var additionalDocument = solution.GetAdditionalDocument(textDocumentIdentifier);
+                if (additionalDocument != null)
+                {
+                    return new ProjectOrDocumentId(additionalDocument.Id);
+                }
+
                 return null;
```

Comment:
does this indicate a bug?

Reply:
Nope!  It is a valid scenario when a document is removed / deleted for example.